### ISSUE TITLE
Resolve IDEA violations from Teamcity 2022.04 (Bundled IntelliJ IDEA 2022.1.3)

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#11865]
https://teamcity.jetbrains.com/buildConfiguration/Checkstyle_IdeaInspectionsMaster/3892633?buildTab=Inspection

Java/Code maturity
[Deprecated member is still used (1)](https://teamcity.jetbrains.com/viewLog.html?tab=Inspection&buildId=3892633&embedded=true#)
Java/Javadoc
[Link specified as plain text (20)](https://teamcity.jetbrains.com/viewLog.html?tab=Inspection&buildId=3892633&embedded=true#)
Java/Verbose or redundant code constructs
[Multiple occurrences of the same expression (2)](https://teamcity.jetbrains.com/viewLog.html?tab=Inspection&buildId=3892633&embedded=true#)
Markdown
[Incorrect table formatting (1)](https://teamcity.jetbrains.com/viewLog.html?tab=Inspection&buildId=3892633&embedded=true#)
RegExp
[Regular expression can be simplified (36)](https://teamcity.jetbrains.com/viewLog.html?tab=Inspection&buildId=3892633&embedded=true#)

**Comment by pbludov** [comment]
The last one is dubious:
![изображение](https://user-images.githubusercontent.com/900805/178112176-aa0f1a28-3aac-4c89-901d-b9d194ab23cf.png)

Maybe `'\n'` will be suitable here.

**Comment by romani** [comment]
> Dmitry Treskunov (TeamCity)                                                                    Jul 8, 2022, 19:31 GMT+3
Hi, I've just installed 2021.2.3, could you please try again?
Dmitry Treskunov, TeamCity Team Lead

**Comment by romani** [comment]
This ticket will be good task for migration of TC from UI to config.
to show how we can smoootly fix all violations and update to new Inspection in same PR.

**Comment by romani** [comment]
Sonar plugin execution is hanging https://teamcity-support.jetbrains.com/hc/en-us/requests/4153345

**Comment by romani** [comment]
Student can download from Jetbrains exactly this version of IDEA, and run all on local.

**Comment by Vyom-Yadav** [comment]
Getting a 404 on all links

**Comment by nick-mancuso** [comment]
Yes, whatever build they were obtained from has probably been removed. We can just run it locally to get the same result.

**Comment by Vyom-Yadav** [comment]
https://github.com/checkstyle/checkstyle/blob/1c92749793f74d524e5f6801e47ed0ec880af4fc/src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java#L89-L102

@nick-mancuso @pbludov @romani 
The method is used in unit testing, are we planning to remove it?

**Comment by nick-mancuso** [comment]
>The method is used in unit testing, are we planning to remove it?

No, we must suppress this for now. We must maintain coverage until it is removed.
